tests/python_functional/src/syslog_ng_config/statements/sources/source_writer.py

```diff
@@ -24,22 +24,20 @@
 logger = logging.getLogger(__name__)
 
 
-class SourceDriver(object):
-    group_type = "source"
+class SourceWriter(object):
+    def __init__(self, driver_io_cls):
+        self.driver_io_cls = driver_io_cls
+        self.actual_driver_resource = None
+        self.driver_io = None
 
-    def __init__(self, IOClass, positional_parameters=[], options={}):
-        self.__IOClass = IOClass
-        self.__writer = None
-        self.positional_parameters = positional_parameters
-        self.options = options
+    def construct_writer(self, driver_resource):
```

Comment:
This should be called constructor_parameters, and should be handled as a list. Or better, this method should be pass, and overridden by a FileSourceWriter.